src/poetry/utils/helpers.py

```diff
@@ -312,3 +336,25 @@ def extractall(source: Path, dest: Path, zip: bool) -> None:
                 archive.extractall(dest, filter="data")
             else:
                 archive.extractall(dest)
+
+
+def get_highest_priority_hash_type(
```

Comment:
Wants a (parameterized) test in `test_helpers` to cover the different cases.

Reply:
That's still open. I know it is implicitly (partially) tested in `test_executor.py`. Nevertheless, we should add an explicit test for this function in `test_helpers.py`.